models/snapchat__utm_report.sql

```diff
@@ -52,24 +57,25 @@ with report as (
         creatives.utm_campaign,
         creatives.utm_content,
         creatives.utm_term,
-        sum(report.spend) as spend,
-        sum(report.impressions) as impressions,
-        sum(report.swipes) as swipes
-    from report
+        sum(ad_hourly.swipes) as swipes,
+        sum(ad_hourly.impressions) as impressions,
+        round(sum(ad_hourly.spend)) as spend
+    
+    from ad_hourly
     left join ads 
-        on report.ad_id = ads.ad_id
-    left join creatives
-        on ads.creative_id = creatives.creative_id
+        on ad_hourly.ad_id = ads.ad_id
     left join ad_squads
         on ads.ad_squad_id = ad_squads.ad_squad_id
     left join campaigns
         on ad_squads.campaign_id = campaigns.campaign_id
-    left join accounts
-        on campaigns.ad_account_id = accounts.ad_account_id
-    {{ dbt_utils.group_by(17) }}
-
+    left join account
+        on campaigns.ad_account_id = account.ad_account_id
+    left join creatives
+        on ads.creative_id = creatives.creative_id
+    
+    {{ dbt_utils.group_by(21) }}
```

Comment:
We decided that the url report should contain _only_ ads that utilize utm parameters. Can you follow the same modeling logic used in Google Ads to filter out the non url ads. All ads are already captured in the ad_report.

https://github.com/fivetran/dbt_google_ads/blob/8c2240828ed58067d5ecd32203652cf606bf446a/models/google_ads__url_report.sql#L86-L88